tests/test_deserialization.py

```diff
@@ -1387,6 +1387,23 @@ class Meta:
             MySchema(unknown=DROP).load({'foo': 3, 'bar': 5}, unknown=RAISE)
 
 
+    def test_unknown_fields_deserialization_with_data_key(self):
+        class MySchema(Schema):
+            foo = fields.Integer(data_key="Foo")
+
+        data = MySchema().load({"Foo": 1})
+        assert data["foo"] == 1
+        assert "Foo" not in data
+
+        data = MySchema(unknown=RAISE).load({"Foo": 1})
+        assert data["foo"] == 1
+        assert "Foo" not in data
```

Comment:
Minor: Can you also add an assertion that `foo` is considered an unknown field and `MySchema(unknown=RAISE).load({"foo": 1})` should therefore raise an error?

Reply:
Done